deviceModeInit.js

```diff
@@ -379,6 +459,60 @@ var rudderTracking = (function () {
     return defer.promise();
   }
 
+  // utility function to get cookie value
+  function cookie_parse() {
+    var obj = {};
+    var pairs = document.cookie.split(/ *; */);
+    var pair;
+    if ("" == pairs[0]) return obj;
+    for (var i = 0; i < pairs.length; ++i) {
+      pair = pairs[i].split("=");
+      obj[pair[0]] = pair[1];
+    }
+    return obj;
+  }
+
+  function cookie_action(agr = {}) {
+    let output;
+    const {
+      name,
+      value,
+      expire_hr,
+      path = "/",
+      samesite = "Lax",
+      action = "get",
+    } = agr;
+    let expires = "";
+    if (expire_hr) {
+      const date = new Date();
+      date.setTime(date.getTime() + expire_hr * 60 * 60 * 1000);
+      expires = "; expires=" + date.toUTCString();
+    }
+
+    switch (action) {
+      case "set":
+        document.cookie =
+          name +
+          "=" +
+          value +
+          expires +
+          ";path=" +
+          path +
+          ";SameSite=" +
+          samesite;
+        output = "Cookie Successfully Set";
+        break;
+      case "get":
+        let cookieObj = cookie_parse();
+        output = cookieObj[name];
+        break;
+      default:
+        output = "Invalid Action";
```

Comment:
Shouldn't it return a value indicating failure? (`undefined` or `null`)

Reply:
We are expecting an object in case of output for get situation. Hence I returned string, for set situation we are not expecting a response right now

Reply:
But this is beyond `set` and `get` operations. Anyway, I think a simple `getCookie` and `setCookie` functions would be ideal.